File: an2/Semestrul1/Metode Avansate de Programare/Seminar8/src/main/java/domain/Task.java
package domain;

import java.util.Objects;

public abstract class Task extends Entity<String>{

    private String description;

    public Task(String taskId, String description) {
        super.setId(taskId);
        this.description = description;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return super.getId() + " " + description;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Task)) return false;
        Task task = (Task) o;
        return Objects.equals(super.getId(), task.getId()) &&
                Objects.equals(getDescription(), task.getDescription());
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.getId(), getDescription());
    }

    public abstract void execute();
}

File: an2/Semestrul1/Metode Avansate de Programare/sem11_12/model/Task.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sem11_12.Model
{
    abstract class Task : Entity<string>
    {
        private string description;

        public Task()
        {
        }


        public string ID { get; set ; }
        public string Description { get => description; set => description = value; }

        public abstract void Execute();

        public override string ToString()
        {
            return "id=" + ID + "|description=" + Description;
        }
    }
}
